app/portainer/views/stacks/edit/stackController.js

```diff
@@ -372,6 +372,15 @@ angular.module('portainer.app').controller('StackController', [
         var stackId = $transition$.params().id;
         loadStack(stackId);
       }
+
+      try {
+        const endpoint = await EndpointService.endpoint($scope.currentEndpointId);
```

Comment:
same as the previous comment, just load from the resolver (same as loading a service)